packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -497,14 +637,29 @@ class MemoryTimeline {
   }
 }
 
+/// Supports saving and loading memory samples.
 class MemoryLog {
   MemoryLog(this.controller);
 
   MemoryController controller;
 
   /// Persist the the live memory data to a JSON file in the /tmp directory.
   void exportMemory() {
-    final liveData = controller.memoryTimeline.data;
+    final liveData = controller.memoryTimeline.liveData;
+
+    bool pseudoData = false;
+    if (liveData.isEmpty) {
+      // Used to create empty memory log for test.
+      pseudoData = true;
+      liveData.add(HeapSample(
+        DateTime.now().microsecondsSinceEpoch,
+        0,
+        0,
+        0,
+        0,
+        false,
+      ));
```

Comment:
is there anyway to mock this in the test instead of the controller?

Reply:
Not yet there's TODO's mentioning this in the the memory_screen_test and the MemoryLog too.